src/chat/zulip.md

```diff
@@ -0,0 +1,180 @@
+users should feel free to create groups as needed, though it is currently
+expected that this is somewhat rare. You should name your group similar to how
+you would name a stream for the same purpose, though groups can be more
+fine-grained (or less). For example, `@T-compiler/meeting` currently does not
+have a dedicated stream.
+
+## Appropriate conversation
+
+In most streams, you should try to keep conversations related to team business.
+The `#general` stream is a bit broader, but even there, discussions should be
+closely related to Rust (though may not relate to projects of any particular
+team). All channels are expected to be used for discussions related to the Rust
+project, though; discussions of (for example) wildlife or sightseeing are not
+appropriate.
+
+## Streams
+
+These are similar to "channels" on other platforms (i.e., there should not be
+too many). On the other hand, you can choose which streams you subscribe to, so
+there can be more than channels on other platforms. Read [Zulip's
+documentation](https://zulipchat.com/help/about-streams-and-topics) for more
+details.
+
+Streams are appropriate for any Rust official group. For example, working
+groups, project groups, teams are all examples of official groups. These should
+ideally also be represented in the [team repository](../infra/team-maintenance.md).
+
+### Default streams
+
+The default set of streams is chosen to allow incoming people to be able to have
+at least one place to go that can then, if necessary, direct them to a more
+specific location.
+
+Currently that means that every top-level group present on Zulip is by default
+visible. Specifically, no stream that contains a `/` will be enabled by default.
+
+Currently this set is:
+ * general
+ * t-lang
+ * t-compiler
+ * project-ffi-unwind
+ * project-inline-asm
+ * project-safe-transmute
+ * rust-survey-2019
+ * wg-async-foundations
+ * wg-database
+ * wg-formal-methods
+ * wg-secure-code
+ * wg-traits
+ * zulip
+
+FIXME: this may be too many -- in particular, I'm not sure how I'd justify this,
+but maybe a better default set is (#general, #zulip, #announce, #new members),
+with the expected getting started workflow for everyone being to setup
+subscribed streams.
+
+### Stream naming
+
+A stream should be named such as `#t-{team}/{group name}`. For example,
+`#t-compiler/wg-parallel-rustc`. More levels of nesting are fine, e.g., a
+working group might want "subgroups" as well, though you may want to omit the
+team name in such a case -- keeping the stream name short is good for usability,
+to avoid confusion between different streams which share the same prefix.
+
+If no top-level team exists, or the group spans multiple teams (e.g.,
+project-ffi-unwind), then the top level team should be omitted.
+
+It is not generally appropriate to have groups that do not correspond to a set
+of people. For example, `#t-compiler/help` would not be appropriate, as such
```

Comment:
This still seems like something to resolve

Reply:
Yes, I think I've been uncertain myself. I actually thought I had removed this language in the latest draft, but having thought about it a bit more, I think my main problem is that I think if we *do* have it, it should be explicitly for "I am a rustc contributor" type help, not more general help -- i.e., I would like to avoid all channels except (maybe) `#general` becoming a place for "general questions" if that makes sense.

Specifically, I find that it's really productive (at least for me) to not need to see/worry about random users asking questions in the same place as "higher priority" compiler contributors, mainly because they're different frames of mind.

I guess -- to conclude -- I think I want to replace this language, saying instead something along these lines:

Streams should be clearly communicated as being for a specific purpose. That purpose can be broad, but it should likely include a group of some kind (even if that group is transient, e.g., people who are having trouble with the rust build system, or people working on the compiler). Furthermore, we do not currently intend for this Zulip to be a general place for community projects not affiliated with the Rust organization; if they wish to use Zulip, it is [free for open source](https://zulipchat.com/for/open-source/).

How does that sound?

Reply:
Good!